docs/blog/2022-08-12-observing-randomness.md

```diff
@@ -0,0 +1,140 @@
+date: 2022-08-11
+tags: 
+    - Features
+---
+
+_Monitoring distributed systems is a challenge. While there are established patterns and solutions for monitoring the most common types of applications, there is usually no silver bullet. In this blog post we outline the challenges that we faced when implementing monitoring for one such system, drand._
+
+## Observing Randomness
+
+Monitoring distributed systems is a challenge. While there are established patterns and solutions for monitoring the most common types of applications, they usually assume a run-of-the mill application:
+
+- It is web based.
+- All nodes and networks are controlled by a single organization.
+- There is a single centralized repository which stores all nodes’ metrics.
+- Each node knows only of its own metrics and has no view of other nodes’ state.
+
+Therefore, the challenges are multiplied if your application is not web-based, running on nodes and networks that are controlled by diverse organizations and you want to enable each of the participating organizations to have a global view of the application without being tied to a single repository. In this blog post I will outline the challenges that we faced when implementing monitoring for one such system, drand.
+
+## What is drand?
+
+[Drand](https://drand.love/) is a public, verifiable, unpredictable and unbiased randomness beacon. It is based on the idea that, to generate randomness in a reliable and trustworthy way, you cannot depend on a single entity to control the infrastructure where it runs. For this, we have sought the collaboration of multiple organizations to run their own drand daemons, forming the [League of Entropy](https://blog.cloudflare.com/league-of-entropy/) (*LoE*) - a distributed organization where each of the participating partners runs a drand node. There are other resources, such as the project’s [documentation pages](https://drand.love/docs/), which explain the cryptographic details regarding how randomness is generated by multiple parties using threshold cryptography. This blog post will instead concentrate on the challenges we faced when we tried to design, implement and deploy monitoring infrastructure for this distributed network.
+
+As a little bit of background, a drand *network* is composed of multiple *nodes*. Each network generates one (or more) *beacons*, each of which generates separate randomness beacons at a specific interval, called a *round*. The number of nodes associated with each beacon is fixed at any given point in time and is only changed during a *ceremony*. During the ceremony a *threshold* is set, which is the minimum number of nodes that need to be connected to the network at any given time for randomness generation to succeed.
```

Comment:
@AnomalRoil how critical would you say that this change is? I was going for a higher-level description, going from the network down to the nodes instead of from the nodes up to the network.

Reply:
Well my main issue is that IMO "Each network generates one (or more) *beacons*, each of which generates separate randomness beacons" is wrong. 

A network doesn't generate beacons, or it does so if we mean the rounds themselves.
And saying that a beacon generates a beacon seems weird.

Maybe instead?
```suggestion
As a little bit of background, a drand *network* is composed of multiple *nodes*. A given network is also sometimes called a *beacon chain*, since each network generates separate randomness beacons at a specific interval, called also a *round*. The number of nodes on a network is fixed at any given point in time and is only changed during a *ceremony*. During the ceremony a *threshold* is set, which is the minimum number of nodes that need to be connected to the network at any given time for randomness generation to succeed.
```